aptos-move/framework/aptos-framework/sources/fungible_asset.move

```diff
@@ -120,6 +122,10 @@ module aptos_framework::fungible_asset {
         project_uri: String,
     }
 
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// Defines a fungible asset stores as being untransferable at the object layer.
+    struct Untransferable has key {}
```

Comment:
Can we use an enum style with another struct `Transferrable` mutually exclusive with `Untransferable` so `TransferRef` and `Untransferrable` object are also mutually exclusive?
if `Untransferable` exists, no `TransferRef` will be generated.
if `TransferRef` is generated, `Transferable` will exists and you cannot make an object `Untransferrable`. This way we don't have to `exists<Transferrable>` in the following function calls because the `enable_ungated_transfer` is always false and we are ensured `TransferRef` does not exist for this object.

Reply:
I think this may be cleaner.

Reply:
That doesn't work, since either can be generated if given a `ConstructorRef` prior to calling `set_untransferrable`.

Reply:
whoever gets called first will set a flag there which prohibits the other. So it should work